### PR TITLE
Add user and password to dsn string

### PR DESCRIPTION
**Pull request by IceShack**
This should fix issue #311 aswell

### CODE REVIEW

File: Command/AbstractCommand.php [hunk 1]
```diff
@@ -270,6 +270,14 @@ protected function getConnectionData($name)
...
+        // Add user and password to dsn string
+        $dsn = explode(';', $connection['dsn']);
+        if (isset($connection['user'])) {
+            $dsn[] = 'user=' . urlencode($connection['user']);
+        }
+        if (isset($connection['password'])) {
+            $dsn[] = 'password=' . urlencode($connection['password']);
+        }
 
         return $connection['dsn'];
```
Comment (havvg) [inline Command/AbstractCommand.php]: That's the wrong place to modify the DSN. The DSN is vendor specific, so it should be set correctly when read here.

**Comment by havvg** [comment]
You should provide the correct DSN using the configuration, including username and password.

**Comment by IceShack** [comment]
The DSN gets converted back into the connection config here:
https://github.com/propelorm/Propel2/blob/master/src/Propel/Generator/Command/AbstractCommand.php#L118

So even a vendor specific dsn should be parsed properly.

OR do you mean I should add user/password in my sf configs dsn?
e.g.:

    propel:
        database:
            connections:
                default:
                    adapter:    "pdo_mysql"
                    dsn:        "pdo_mysql:host=localhost;dbname=acme;charset=UTF8;user=test;password=test"

**Comment by havvg** [comment]
Yes, I mean setting it in the config, as you pasted.

**Comment by marcj** [comment]
Well, we give the user the ability to configure username and password under an extra property, so we should just make it workable.

**Comment by havvg** [comment]
Definitely, but not by managing it within the commands. The DSN should be correct after the configuration has been processed. So the whole system may use the DSN as it without any modification.

**Comment by IceShack** [comment]
So add user/password while processing the sf configuration options in the PropelExtension.php?

**Comment by marcj** [comment]
> Definitely, but not by managing it within the commands. The DSN should be correct after the configuration has been processed. So the whole system may use the DSN as it without any modification.

A DSN for PHP's PDO is correct also without username and password. For username/password you have in PDO an extra constructor argument. The issue here is that we can not pass 3 values at once from PropelBundle wrapper commands to our Propel commands for one argument. I don't see any problem using the approach @IceShack gave to us.

If you don't want to see it within commands, please provide @IceShack another way to implement this.

**Comment by marcj** [comment]
ping @havvg

**Comment by havvg** [comment]
I will merge this for now, but we should change the behavior to a more decoupled way to work with the DSN in any place.

**Pull request merged**